### ISSUE TITLE
Role "Button" is not announced by VoiceOver for every Span that has a GestureRecognizer

### ISSUE THREAD

**Issue by ChristopherStephan** [dotnet/maui#25764]
### Description

When having a Label with multiple Spans that have a GestureRecognizer attached, the screen reader VoiceOver from iOS will not announce the role "Button" for every Span. It will only announce the role "Button" once.

```
<Label>
    <Label.FormattedText>
        <FormattedString>
            <Span Text="By selecting 'Agree and get started', you agree to our " />
            <Span Text="End User License Agreement"
                    TextDecorations="Underline">
                <Span.GestureRecognizers>
                    <TapGestureRecognizer Command="{Binding TapCommand}"
                                            CommandParameter="https://learn.microsoft.com/dotnet/maui/"
                                            />
                </Span.GestureRecognizers>
            </Span>
            <Span Text=" and acknowledge the " />
            <Span Text="Privacy Policy"
                    TextDecorations="Underline">
                <Span.GestureRecognizers>
                    <TapGestureRecognizer Command="{Binding TapCommand}"
                                            CommandParameter="https://learn.microsoft.com/dotnet/maui/" />
                </Span.GestureRecognizers>
            </Span>
        </FormattedString>
    </Label.FormattedText>
</Label>
```

Actual outcome: "By selecting 'Agree and get started', you agree to our End User License Agreement and acknowledge the Privacy Policy. **Button**"
Expected outcome: "By selecting 'Agree and get started', you agree to our End User License Agreement **Button** and acknowledge the Privacy Policy. **Button**"

### Version with bug

8.0.40 SR5

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Affected platforms

iOS

### Relevant log output

```shell

```
